### ISSUE TITLE
NullPointerException in UnresolvedOfflinePlayer

### ISSUE THREAD

**Issue by kh498** [bensku/Skript#237]
For some reason I get an error with UnresolvedOfflinePlayer. I looked in [the file](https://github.com/bensku/Skript/blob/master/src/main/java/ch/njol/skript/bukkitutil/UnresolvedOfflinePlayer.java#L62) that you can just add a catch for a NullPointerException to suppress the error. 😈 

  ```
  [23:09:12 INFO]: kh498 [0e8adf51-dc94-3477-bd6c-931cabeed7cc]: /enum
  [23:09:16 ERROR]: #!#!
  [23:09:16 ERROR]: #!#! [Skript] Severe Error:
[23:09:16 ERROR]: #!#! Exception in thread Skript offline player resolver thread (fetches UUIDs from the minecraft servers)
[23:09:16 ERROR]: #!#!
[23:09:16 ERROR]: #!#! If you're developing an add-on for Skript this likely means that you have done something wrong.
[23:09:16 ERROR]: #!#! If you're a server admin however please go to https://github.com/bensku/Skript/issues/
[23:09:16 ERROR]: #!#! and check whether this error has already been reported.
[23:09:16 ERROR]: #!#! If not please create a new ticket with a meaningful title, copy & paste this whole error into it (or use paste service),
[23:09:16 ERROR]: #!#! and describe what you did before it happened and/or what you think caused the error.
[23:09:16 ERROR]: #!#! If you think that it's a trigger that's causing the error please post the trigger as well.
[23:09:16 ERROR]: #!#! By following this guide fixing the error should be easy and done fast.
[23:09:16 ERROR]: #!#!
[23:09:16 ERROR]: #!#! Stack trace:
[23:09:16 ERROR]: #!#! java.lang.NullPointerException
[23:09:16 ERROR]: #!#!     at ch.njol.skript.bukkitutil.UnresolvedOfflinePlayer$1.run(UnresolvedOfflinePlayer.java:59)
[23:09:16 ERROR]: #!#!     at java.lang.Thread.run(Unknown Source)
[23:09:16 ERROR]: #!#!
[23:09:16 ERROR]: #!#! Version Information:
[23:09:16 ERROR]: #!#!   Skript: 2.2-dev20
[23:09:16 ERROR]: #!#!   Bukkit: 1.10.2-R0.1-SNAPSHOT
[23:09:16 ERROR]: #!#!   Minecraft: 1.10.2
[23:09:16 ERROR]: #!#!   Java: 1.8.0_112 (Java HotSpot(TM) 64-Bit Server VM 25.112-b15)
[23:09:16 ERROR]: #!#!   OS: Windows 10 amd64 10.0
[23:09:16 ERROR]: #!#!
[23:09:16 ERROR]: #!#! Running CraftBukkit: false
[23:09:16 ERROR]: #!#!
[23:09:16 ERROR]: #!#! Current node: null
[23:09:16 ERROR]: #!#! Current item: null
[23:09:16 ERROR]: #!#!
[23:09:16 ERROR]: #!#! Thread: Skript offline player resolver thread (fetches UUIDs from the minecraft servers)
[23:09:16 ERROR]: #!#!
[23:09:16 ERROR]: #!#! End of Error.
[23:09:16 ERROR]: #!#!
```

**Comment by bensku** [comment]
No, I'm not going to ignore this error. It needs to "crash" if it doesn't work, otherwise much worse things might happen. Including providing wrong data to scripts and possible variable corruption.

This is strange bug, because it almost seema that static final field which IS initialized would be null. Weird.

**Comment by Weefle** [comment]
Please try again to resolve it! :/

**Comment by bensku** [comment]
Attempting a fix, can't say if will be successful.

**Comment by bensku** [comment]
Wait what Github. I definitely did not *close* this.

**Comment by xXAndrew28Xx** [comment]
Maybe it was smart and closed this because you wrote `fix #237`

**Comment by Namnodorel** [comment]
Yes, Github automatically closes issues if you reference it with certain keywords in a commit

**Comment by bensku** [comment]
Hm. I would rather wait for user to confirm that it was fixed, or at least for the release to be uploaded. But anyway...

**Comment by Weefle** [comment]
The bug is still there!
2.2-dev21c

**Comment by bensku** [comment]
Yes, but what error message do you get? I "improved error reporting" for this, since last error message was basically saying "Java standard thread API returns null", which is not likely.

**Comment by Weefle** [comment]
When I reload the latest spigot-1.11 and skript 2.2-dev21c, I get this error after a while: http://hastebin.com/upucacutex.md

**Comment by bensku** [comment]
Yes, thanks! Now I can try to fix this...

**Comment by Weefle** [comment]
And... not fixed in 2.2-dev22b: http://hastebin.com/becupivawo.md

**Comment by melinstagibson** [comment]
Would this actually solve the treatment of playernames compared to uuids or
would it still be like var.%oldname% =\= var.%newname%?

Am 27.11.2016 7:32 nachm. schrieb "Weefle" <notifications@github.com>:

**Comment by Weefle** [comment]
I don't know :/

**Comment by bensku** [comment]
No, this wouldn't solve that. Probably.

**Comment by AbstractSpectrum** [comment]
As of v2.2-dev22f error is persisting.

https://hastebin.com/nucaxetuke.md

**Comment by bensku** [comment]
Attempting some kind of fix for next release. I still have no idea why this happens for some people but not others.

**Comment by Weefle** [comment]
I can confirm this error again in the latest release: v2.2-dev22g
:/

**Comment by kh498** [comment]
Just a side note that might help, I am running my server in offline mode.

**Comment by RoyCurtis** [comment]
FWIW, I accidentally found a reliable reproduction for this, whilst testing another issue:

*Tested with only Skript v2.2-dev22g on Spigot `7d78b81-f9e0440` and Paper b993*

1. Start the server
1. Wait for up to 15 seconds after loading is done. Observe there are no Skript exceptions
1. Execute `reload confirm` to reload the server
1. Wait for up to 15 seconds after loading is done. [Observe that a Skript NPE eventually happens](http://i.imgur.com/2LBC6w3.png)

I realize that server reloading is unsupported, but I have not seen this being asked here: those who are getting this issue, was this after a `/reload` or from cold server startup?

**Comment by Syst3ms** [comment]
/reload has always caused a shit ton of isuues to me. I've not encountered those, but i know two others that are still very annoying and severe : 
* Doing a /reload makes Skript unable to load a script properly. Skript will not detect changes on a script in that case.
* You can't reload anything. And I mean anything. Throws an error in console each time.

**Comment by TheBentoBox** [comment]
You probably don't even need to do an additional restart where you remove Skript and all of the addons, just giving the server a full and proper restart should allow everything to register properly and do the job fine.

**Comment by Tuke-Nuke** [comment]
Hi, I want to report this problem to see if it maight be helpful:

Stack trace: https://hastebin.com/sonemovare.md

Infomartion: 
* The error occurred when I tried to stop the server (it's a test server in my own PC). I was starting and stopping many times as I was making some request for your fork (I didn't make any changes, only few expressions added). It only did happen **once**.
* Plugins: TuSKe v1.7-beta.3, SkRambled v2.1, skUtilities v0.7.9. (It is a test development server)
* Running Spigot `git-Spigot-90f61bc-29e1d72`.
* 0 Variables stored.
* 0 Scripts loaded.

**Comment by Goetheus** [comment]
Hi! I come here to report this same error when I stop my server. I can't close it and restart it...
I'm on 1.11.2 spigot server with Skript 2.2 dev24b.

Addons : MundoSk, SharpSk, Skellett, SkQuery, SkRayFall, SkStuff, SkUtilities, TuSKe and Umbaska.

**Important** : I noticed that error happens only with the plugin Dynmap installed and when it does a render!

Thank you for your work and keep going! :D

`[23:54:21 ERROR]: #!#!
[23:54:21 ERROR]: #!#! [Skript] Severe Error:
[23:54:21 ERROR]: #!#! An error happened when trying to enably resolving offline players, so the feature has been turned off. Please report this at https://github.com/bensku/Skript/issues/237
[23:54:21 ERROR]: #!#!
[23:54:21 ERROR]: #!#! If you're developing an add-on for Skript this likely means that you have done something wrong.
[23:54:21 ERROR]: #!#! If you're a server admin however please go to https://github.com/bensku/Skript/issues/
[23:54:21 ERROR]: #!#! and check whether this error has already been reported.
[23:54:21 ERROR]: #!#! If not please create a new ticket with a meaningful title, copy & paste this whole error into it (or use paste service),
[23:54:21 ERROR]: #!#! and describe what you did before it happened and/or what you think caused the error.
[23:54:21 ERROR]: #!#! If you think that it's a trigger that's causing the error please post the trigger as well.
[23:54:21 ERROR]: #!#! By following this guide fixing the error should be easy and done fast.
[23:54:21 ERROR]: #!#!
[23:54:21 ERROR]: #!#! Stack trace:
[23:54:21 ERROR]: #!#! java.lang.NullPointerException: null offline player resolve queue
[23:54:21 ERROR]: #!#!     at ch.njol.skript.bukkitutil.UnresolvedOfflinePlayer$1.run(UnresolvedOfflinePlayer.java:58)
[23:54:21 ERROR]: #!#!     at java.lang.Thread.run(Unknown Source)
[23:54:21 ERROR]: #!#!
[23:54:21 ERROR]: #!#! Version Information:
[23:54:21 ERROR]: #!#!   Skript: 2.2-dev24b
[23:54:21 ERROR]: #!#!   Bukkit: 1.11.2-R0.1-SNAPSHOT
[23:54:21 ERROR]: #!#!   Minecraft: 1.11.2
[23:54:21 ERROR]: #!#!   Java: 1.8.0_77 (Java HotSpot(TM) 64-Bit Server VM 25.77-b03)
[23:54:21 ERROR]: #!#!   OS: Windows 10 amd64 10.0
[23:54:21 ERROR]: #!#!
[23:54:21 ERROR]: #!#! Running CraftBukkit: false
[23:54:21 ERROR]: #!#!
[23:54:21 ERROR]: #!#! Current node: null
[23:54:21 ERROR]: #!#! Current item: null
[23:54:21 ERROR]: #!#!
[23:54:21 ERROR]: #!#! Thread: Skript offline player resolver thread (fetches UUIDs from the minecraft servers)
[23:54:21 ERROR]: #!#!
[23:54:21 ERROR]: #!#! End of Error.
[23:54:21 ERROR]: #!#!`

**Comment by Syst3ms** [comment]
I think @bensku is going to cry

**Comment by bensku** [comment]
I reworked that again, let's see if it helps with next version. I also fixed other problems with offline players.

**Comment by supertassu** [comment]
Can confirm at latest version (dev25) & Spigot 1.11.2.

**Comment by Remixful** [comment]
I am still getting this error as of Skript version is 2.2-dev25

**Comment by bensku** [comment]
I'm confident that this has something to do with /reload, or similar effects. However, "similar effects" might include server shutting down, so it is a bit more complicated...

**Comment by FilipeNock** [comment]
Same error.....

**Comment by Shane1199** [comment]
This bug is still happening in the newest version of Skript, I was on 1.7.10 with the protocol hack.

**Comment by Guarmanda** [comment]
still happening, but not on server start, only after two reloads for me. skript 2.2 dev 27 and 1.11 without protocol hack.
`[14:13:24] [Skript offline player resolver thread (fetches UUIDs from the minecraft servers)/ERROR]: #!#! 
[14:13:24] [Skript offline player resolver thread (fetches UUIDs from the minecraft servers)/ERROR]: #!#! [Skript] Severe Error:
[14:13:24] [Skript offline player resolver thread (fetches UUIDs from the minecraft servers)/ERROR]: #!#! An error happened when trying to enably resolving offline players, so the feature has been turned off. Please report this at https://github.com/bensku/Skript/issues/237
[14:13:24] [Skript offline player resolver thread (fetches UUIDs from the minecraft servers)/ERROR]: #!#! 
[14:13:24] [Skript offline player resolver thread (fetches UUIDs from the minecraft servers)/ERROR]: #!#! If you're developing an add-on for Skript this likely means that you have done something wrong.
[14:13:24] [Skript offline player resolver thread (fetches UUIDs from the minecraft servers)/ERROR]: #!#! If you're a server admin however please go to https://github.com/bensku/Skript/issues/
[14:13:24] [Skript offline player resolver thread (fetches UUIDs from the minecraft servers)/ERROR]: #!#! and check whether this error has already been reported.
[14:13:24] [Skript offline player resolver thread (fetches UUIDs from the minecraft servers)/ERROR]: #!#! If not please create a new ticket with a meaningful title, copy & paste this whole error into it (or use paste service),
[14:13:24] [Skript offline player resolver thread (fetches UUIDs from the minecraft servers)/ERROR]: #!#! and describe what you did before it happened and/or what you think caused the error.
[14:13:24] [Skript offline player resolver thread (fetches UUIDs from the minecraft servers)/ERROR]: #!#! If you think that it's a trigger that's causing the error please post the trigger as well.
[14:13:24] [Skript offline player resolver thread (fetches UUIDs from the minecraft servers)/ERROR]: #!#! By following this guide fixing the error should be easy and done fast.
[14:13:24] [Skript offline player resolver thread (fetches UUIDs from the minecraft servers)/ERROR]: #!#! 
[14:13:24] [Skript offline player resolver thread (fetches UUIDs from the minecraft servers)/ERROR]: #!#! Stack trace:
[14:13:24] [Skript offline player resolver thread (fetches UUIDs from the minecraft servers)/ERROR]: #!#! java.lang.NullPointerException: null offline player resolve queue
[14:13:24] [Skript offline player resolver thread (fetches UUIDs from the minecraft servers)/ERROR]: #!#!     at ch.njol.skript.bukkitutil.UnresolvedOfflinePlayer$1.run(UnresolvedOfflinePlayer.java:59)
[14:13:24] [Skript offline player resolver thread (fetches UUIDs from the minecraft servers)/ERROR]: #!#!     at java.lang.Thread.run(Unknown Source)
[14:13:24] [Skript offline player resolver thread (fetches UUIDs from the minecraft servers)/ERROR]: #!#! 
[14:13:24] [Skript offline player resolver thread (fetches UUIDs from the minecraft servers)/ERROR]: #!#! Version Information:
[14:13:24] [Skript offline player resolver thread (fetches UUIDs from the minecraft servers)/ERROR]: #!#!   Skript: 2.2-dev27
[14:13:24] [Skript offline player resolver thread (fetches UUIDs from the minecraft servers)/ERROR]: #!#!   Bukkit: 1.11.2-R0.1-SNAPSHOT
[14:13:24] [Skript offline player resolver thread (fetches UUIDs from the minecraft servers)/ERROR]: #!#!   Minecraft: 1.11.2
[14:13:24] [Skript offline player resolver thread (fetches UUIDs from the minecraft servers)/ERROR]: #!#!   Java: 1.8.0_121 (Java HotSpot(TM) Client VM 25.121-b13)
[14:13:24] [Skript offline player resolver thread (fetches UUIDs from the minecraft servers)/ERROR]: #!#!   OS: Windows 10 x86 10.0
[14:13:24] [Skript offline player resolver thread (fetches UUIDs from the minecraft servers)/ERROR]: #!#! 
[14:13:24] [Skript offline player resolver thread (fetches UUIDs from the minecraft servers)/ERROR]: #!#! Running CraftBukkit: false
[14:13:24] [Skript offline player resolver thread (fetches UUIDs from the minecraft servers)/ERROR]: #!#! 
[14:13:24] [Skript offline player resolver thread (fetches UUIDs from the minecraft servers)/ERROR]: #!#! Current node: null
[14:13:24] [Skript offline player resolver thread (fetches UUIDs from the minecraft servers)/ERROR]: #!#! Current item: null
[14:13:24] [Skript offline player resolver thread (fetches UUIDs from the minecraft servers)/ERROR]: #!#! 
[14:13:24] [Skript offline player resolver thread (fetches UUIDs from the minecraft servers)/ERROR]: #!#! Thread: Skript offline player resolver thread (fetches UUIDs from the minecraft servers)
[14:13:24] [Skript offline player resolver thread (fetches UUIDs from the minecraft servers)/ERROR]: #!#! 
[14:13:24] [Skript offline player resolver thread (fetches UUIDs from the minecraft servers)/ERROR]: #!#! End of Error.
[14:13:24] [Skript offline player resolver thread (fetches UUIDs from the minecraft servers)/ERROR]: #!#! `

**Comment by TheBentoBox** [comment]
You shouldn't be using reload, nor reporting errors you get after using /reload. It's not really supported by Skript and it's an unsafe thing to do in general.

**Comment by v3Monsta** [comment]
Started a fresh install of a server running PaperMC (git-paper-1147) and installed the latest Skript (2.2-dev29) and had this error occur during the server shutdown process, mainly the saving world process which hung for about 5 seconds before Skript displayed this error.

Log: https://hastebin.com/kixomasuwa.md

Restarting the server ended with the same error at the same time.

**Comment by Nariko13** [comment]
I am just doing what the console tell me to do, reporting the error to the link given, don't pay attention to me ^^"

[12:50:53] [Skript offline player resolver thread (fetches UUIDs from the minecraft servers)/ERROR]: #!#! 

[12:50:53] [Skript offline player resolver thread (fetches UUIDs from the minecraft servers)/ERROR]: #!#! [Skript] Severe Error:

[12:50:53] [Skript offline player resolver thread (fetches UUIDs from the minecraft servers)/ERROR]: #!#! An error happened when trying to enably resolving offline players, so the feature has been turned off. Please report this at https://github.com/bensku/Skript/issues/237

[12:50:53] [Skript offline player resolver thread (fetches UUIDs from the minecraft servers)/ERROR]: #!#! 

[12:50:53] [Skript offline player resolver thread (fetches UUIDs from the minecraft servers)/ERROR]: #!#! If you're developing an add-on for Skript this likely means that you have done something wrong.

[12:50:53] [Skript offline player resolver thread (fetches UUIDs from the minecraft servers)/ERROR]: #!#! If you're a server admin however please go to https://github.com/bensku/Skript/issues/

[12:50:53] [Skript offline player resolver thread (fetches UUIDs from the minecraft servers)/ERROR]: #!#! and check whether this error has already been reported.

[12:50:53] [Skript offline player resolver thread (fetches UUIDs from the minecraft servers)/ERROR]: #!#! If not please create a new ticket with a meaningful title, copy & paste this whole error into it (or use paste service),

[12:50:53] [Skript offline player resolver thread (fetches UUIDs from the minecraft servers)/ERROR]: #!#! and describe what you did before it happened and/or what you think caused the error.

[12:50:53] [Skript offline player resolver thread (fetches UUIDs from the minecraft servers)/ERROR]: #!#! If you think that it's a trigger that's causing the error please post the trigger as well.

[12:50:53] [Skript offline player resolver thread (fetches UUIDs from the minecraft servers)/ERROR]: #!#! By following this guide fixing the error should be easy and done fast.

[12:50:53] [Skript offline player resolver thread (fetches UUIDs from the minecraft servers)/ERROR]: #!#! 

[12:50:53] [Skript offline player resolver thread (fetches UUIDs from the minecraft servers)/ERROR]: #!#! Stack trace:

[12:50:53] [Skript offline player resolver thread (fetches UUIDs from the minecraft servers)/ERROR]: #!#! java.lang.NullPointerException: null offline player resolve queue

[12:50:53] [Skript offline player resolver thread (fetches UUIDs from the minecraft servers)/ERROR]: #!#!     at ch.njol.skript.bukkitutil.UnresolvedOfflinePlayer$1.run(UnresolvedOfflinePlayer.java:61)

[12:50:53] [Skript offline player resolver thread (fetches UUIDs from the minecraft servers)/ERROR]: #!#!     at java.lang.Thread.run(Thread.java:745)

[12:50:53] [Skript offline player resolver thread (fetches UUIDs from the minecraft servers)/ERROR]: #!#! 

[12:50:53] [Skript offline player resolver thread (fetches UUIDs from the minecraft servers)/ERROR]: #!#! Version Information:

[12:50:53] [Skript offline player resolver thread (fetches UUIDs from the minecraft servers)/ERROR]: #!#!   Skript: 2.2-dev25

[12:50:53] [Skript offline player resolver thread (fetches UUIDs from the minecraft servers)/ERROR]: #!#!   Bukkit: 1.11.2-R0.1-SNAPSHOT

[12:50:53] [Skript offline player resolver thread (fetches UUIDs from the minecraft servers)/ERROR]: #!#!   Minecraft: 1.11.2

[12:50:53] [Skript offline player resolver thread (fetches UUIDs from the minecraft servers)/ERROR]: #!#!   Java: 1.8.0_72 (Java HotSpot(TM) 64-Bit Server VM 25.72-b15)

[12:50:53] [Skript offline player resolver thread (fetches UUIDs from the minecraft servers)/ERROR]: #!#!   OS: Linux amd64 3.16.0-4-amd64

[12:50:53] [Skript offline player resolver thread (fetches UUIDs from the minecraft servers)/ERROR]: #!#! 

[12:50:53] [Skript offline player resolver thread (fetches UUIDs from the minecraft servers)/ERROR]: #!#! Running CraftBukkit: false

[12:50:53] [Skript offline player resolver thread (fetches UUIDs from the minecraft servers)/ERROR]: #!#! 

[12:50:53] [Skript offline player resolver thread (fetches UUIDs from the minecraft servers)/ERROR]: #!#! Current node: null

[12:50:53] [Skript offline player resolver thread (fetches UUIDs from the minecraft servers)/ERROR]: #!#! Current item: null

[12:50:53] [Skript offline player resolver thread (fetches UUIDs from the minecraft servers)/ERROR]: #!#! 

[12:50:53] [Skript offline player resolver thread (fetches UUIDs from the minecraft servers)/ERROR]: #!#! Thread: Skript offline player resolver thread (fetches UUIDs from the minecraft servers)

[12:50:53] [Skript offline player resolver thread (fetches UUIDs from the minecraft servers)/ERROR]: #!#! 

[12:50:53] [Skript offline player resolver thread (fetches UUIDs from the minecraft servers)/ERROR]: #!#! End of Error.

[12:50:53] [Skript offline player resolver thread (fetches UUIDs from the minecraft servers)/ERROR]: #!#!

**Comment by Weefle** [comment]
The error is still there in dev30d with the latest spigot 1.12.1 from Yve's mirror
The error happen at the /reload command: https://hastebin.com/ixitebiqiq.md

**Comment by bensku** [comment]
/reload is unsafe to use with Skript. Don't do that.

**Comment by 7ow** [comment]
I had Skript installed on my server. Not using a host. I installed Skript itself. It worked perfectly but when I installed the addons, it started giving that message.

**Comment by BallisticDev** [comment]
[01:35:16 ERROR]: #!#! [Skript] Severe Error:
[01:35:16 ERROR]: #!#! An error happened when trying to enably resolving offline players, so the feature has been turned off. Please report this at https://github.com/bensku/Skript/issues/237
[01:35:16 ERROR]: #!#!
[01:35:16 ERROR]: #!#! Something went horribly wrong with Skript.
[01:35:16 ERROR]: #!#! This issue is NOT your fault! You can't probably fix it yourself, either.
[01:35:16 ERROR]: #!#! Your Minecraft version or server software appears to be unsupported by Skript (bensku's version).
[01:35:16 ERROR]: #!#! Currently only supported servers are Spigot and its forks for Minecraft 1.9 or newer.
[01:35:16 ERROR]: #!#! Other versions might work, but since you're getting this error message something is NOT working,
[01:35:16 ERROR]: #!#! nor it will work, unless you switch to supported platform.
[01:35:16 ERROR]: #!#! Issue tracker: https://github.com/bensku/Skript/issues (only if you know what you're doing!)
[01:35:16 ERROR]: #!#!
[01:35:16 ERROR]: #!#! Stack trace:
[01:35:16 ERROR]: #!#! java.lang.NullPointerException: null offline player resolve queue
[01:35:16 ERROR]: #!#!     at ch.njol.skript.bukkitutil.UnresolvedOfflinePlayer$1.run(UnresolvedOfflinePlayer.java:59)
[01:35:16 ERROR]: #!#!     at java.base/java.lang.Thread.run(Unknown Source)
[01:35:16 ERROR]: #!#!
[01:35:16 ERROR]: #!#! Version Information:
[01:35:16 ERROR]: #!#!
[01:35:16 ERROR]: #!#! [Skript] Severe Error:
[01:35:16 ERROR]: #!#! Exception in thread Skript offline player resolver thread (fetches UUIDs from the minecraft servers)
[01:35:16 ERROR]: #!#!
[01:35:16 ERROR]: #!#! Something went horribly wrong with Skript.
[01:35:16 ERROR]: #!#! This issue is NOT your fault! You can't probably fix it yourself, either.
[01:35:16 ERROR]: #!#! Your Minecraft version or server software appears to be unsupported by Skript (bensku's version).
[01:35:16 ERROR]: #!#! Currently only supported servers are Spigot and its forks for Minecraft 1.9 or newer.
[01:35:16 ERROR]: #!#! Other versions might work, but since you're getting this error message something is NOT working,
[01:35:16 ERROR]: #!#! nor it will work, unless you switch to supported platform.
[01:35:16 ERROR]: #!#! Issue tracker: https://github.com/bensku/Skript/issues (only if you know what you're doing!)
[01:35:16 ERROR]: #!#!
[01:35:16 ERROR]: #!#! Stack trace:
[01:35:16 ERROR]: #!#! java.lang.IllegalStateException
[01:35:16 ERROR]: #!#!     at ch.njol.skript.Skript.getVersion(Skript.java:188)
[01:35:16 ERROR]: #!#!     at ch.njol.skript.Skript.exception(Skript.java:1332)
[01:35:16 ERROR]: #!#!     at ch.njol.skript.Skript.exception(Skript.java:1192)
[01:35:16 ERROR]: #!#!     at ch.njol.skript.bukkitutil.UnresolvedOfflinePlayer$1.run(UnresolvedOfflinePlayer.java:59)
[01:35:16 ERROR]: #!#!     at java.base/java.lang.Thread.run(Unknown Source)
[01:35:16 ERROR]: #!#!
[01:35:16 ERROR]: #!#! Version Information:

**Comment by BallisticDev** [comment]
help

**Comment by Legoman99573** [comment]
I get the same issue and prevents the server from shutting down, thus forcing me to kill the server proccess.

**Comment by BallisticDev** [comment]
welp its so annoying!

**Comment by BallisticDev** [comment]
[23:57:37 INFO]: [Skript] Enabling Skript v2.1.2
[23:57:37 INFO]: [Skript] Successfully generated the config, the example scripts and the aliases files.
[23:57:37 ERROR]: Error occurred while enabling Skript v2.1.2 (Is it up to date?)
java.lang.NoSuchMethodError: org.bukkit.DyeColor.getData()B
        at ch.njol.skript.util.Color.<clinit>(Color.java:82) ~[?:?]
        at ch.njol.skript.SkriptCommand.<clinit>(SkriptCommand.java:60) ~[?:?]
        at ch.njol.skript.Skript.onEnable(Skript.java:254) ~[?:?]
        at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:264) ~[spigot-1.12.2.jar:git-Spigot-d21162c-61e0c69]
        at org.bukkit.plugin.java.JavaPluginLoader.enablePlugin(JavaPluginLoader.java:337) [spigot-1.12.2.jar:git-Spigot-d21162c-61e0c69]
        at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:403) [spigot-1.12.2.jar:git-Spigot-d21162c-61e0c69]
        at org.bukkit.craftbukkit.v1_12_R1.CraftServer.enablePlugin(CraftServer.java:384) [spigot-1.12.2.jar:git-Spigot-d21162c-61e0c69]
        at org.bukkit.craftbukkit.v1_12_R1.CraftServer.enablePlugins(CraftServer.java:333) [spigot-1.12.2.jar:git-Spigot-d21162c-61e0c69]
        at net.minecraft.server.v1_12_R1.MinecraftServer.t(MinecraftServer.java:422) [spigot-1.12.2.jar:git-Spigot-d21162c-61e0c69]
        at net.minecraft.server.v1_12_R1.MinecraftServer.l(MinecraftServer.java:383) [spigot-1.12.2.jar:git-Spigot-d21162c-61e0c69]
        at net.minecraft.server.v1_12_R1.MinecraftServer.a(MinecraftServer.java:338) [spigot-1.12.2.jar:git-Spigot-d21162c-61e0c69]
        at net.minecraft.server.v1_12_R1.DedicatedServer.init(DedicatedServer.java:272) [spigot-1.12.2.jar:git-Spigot-d21162c-61e0c69]
        at net.minecraft.server.v1_12_R1.MinecraftServer.run(MinecraftServer.java:545) [spigot-1.12.2.jar:git-Spigot-d21162c-61e0c69]
        at java.lang.Thread.run(Unknown Source) [?:?]

**Comment by xXAndrew28Xx** [comment]
This doesn't belong here, create a new issue.

**Comment by BallisticDev** [comment]
What do you mean it doesnt belong here??

**Comment by xXAndrew28Xx** [comment]
This is a closed issue involving a NullPointerException while resolving offline players.

**Comment by BallisticDev** [comment]
Ahh i see

**Comment by TheBentoBox** [comment]
You're also not even using this version of Skript @BallisticDev

**Comment by BallisticDev** [comment]
??

**Comment by Legoman99573** [comment]
you are using the original skript plugin. use this fork instead.

**Comment by BallisticDev** [comment]
This makes no sense

**Comment by BallisticDev** [comment]
Im just trying to get this fixed

**Comment by xXAndrew28Xx** [comment]
You're using the version of Skript, 2.1.2, that was made by njol. This github repository is for the Skript version made by bensku. Try switching to this, by going to [releases](https://github.com/bensku/Skript/releases).

**Comment by BallisticDev** [comment]
Ahhh

**Comment by BallisticDev** [comment]
[18:09:25 INFO]: [Skript] Reloading the config and all scripts...
[18:09:25 INFO]: Loaded a total of 79112 english aliases
[18:09:25 ERROR]: #!#!
[18:09:25 ERROR]: #!#! [Skript] Severe Error:
[18:09:25 ERROR]: #!#! Could not load clan.sk
[18:09:25 ERROR]: #!#!
[18:09:25 ERROR]: #!#! Something went horribly wrong with Skript.
[18:09:25 ERROR]: #!#! This issue is NOT your fault! You probably can't fix it yourself, either.
[18:09:25 ERROR]: #!#! It looks like you are using some plugin(s) that alter how Skript works (addons).
[18:09:25 ERROR]: #!#! Here is full list of them:
[18:09:25 ERROR]: #!#! skRayFall (http://sk.rayfall.net/) skUtilities (https://tim740.github.io/) SkQuery Skellett (https://forums.skunity.com/resources/skellett-the-addon-with-a-beast-name.24/) SharpSK ExertSK TuSKe (github.com/Tuke-Nuke/TuSKe)
[18:09:25 ERROR]: #!#! We could not identify which of those are specially related, so this might also be Skript issue.
[18:09:25 ERROR]: #!#! You should try disabling those plugins one by one, trying to find which one causes it.
[18:09:25 ERROR]: #!#! If the error doesn't disappear even after disabling all listed plugins, it is probably Skript issue.
[18:09:25 ERROR]: #!#! In that case, you will be given instruction on how should you report it.
[18:09:25 ERROR]: #!#! On the other hand, if the error disappears when disabling some plugin, report it to author of that plugin.
[18:09:25 ERROR]: #!#! Only if the author tells you to do so, report it as Skript issue (url below)
[18:09:25 ERROR]: #!#! Issue tracker: https://github.com/bensku/Skript/issues (only if you know what you're doing!)
[18:09:25 ERROR]: #!#!
[18:09:25 ERROR]: #!#! Stack trace:
[18:09:25 ERROR]: #!#! ch.njol.skript.SkriptAPIException: No class info found for lambda
[18:09:25 ERROR]: #!#!     at ch.njol.skript.registrations.Classes.getClassInfo(Classes.java:263)
[18:09:25 ERROR]: #!#!     at ch.njol.skript.lang.SkriptParser.createExprInfo(SkriptParser.java:1786)
[18:09:25 ERROR]: #!#!     at ch.njol.skript.lang.SkriptParser.getExprInfo(SkriptParser.java:1753)
[18:09:25 ERROR]: #!#!     at ch.njol.skript.lang.SkriptParser.parse_i(SkriptParser.java:1501)
[18:09:25 ERROR]: #!#!     at ch.njol.skript.lang.SkriptParser.parse(SkriptParser.java:223)
[18:09:25 ERROR]: #!#!     at ch.njol.skript.lang.SkriptParser.parse(SkriptParser.java:178)
[18:09:25 ERROR]: #!#!     at ch.njol.skript.lang.SkriptParser.parseSingleExpr(SkriptParser.java:491)
[18:09:25 ERROR]: #!#!     at ch.njol.skript.lang.SkriptParser.parseExpression(SkriptParser.java:808)
[18:09:25 ERROR]: #!#!     at ch.njol.skript.lang.SkriptParser.parse_i(SkriptParser.java:1517)
[18:09:25 ERROR]: #!#!     at ch.njol.skript.lang.SkriptParser.parse_i(SkriptParser.java:1447)
[18:09:25 ERROR]: #!#!     at ch.njol.skript.lang.SkriptParser.parse_i(SkriptParser.java:1474)
[18:09:25 ERROR]: #!#!     at ch.njol.skript.lang.SkriptParser.parse_i(SkriptParser.java:1474)
[18:09:25 ERROR]: #!#!     at ch.njol.skript.lang.SkriptParser.parse_i(SkriptParser.java:1474)
[18:09:25 ERROR]: #!#!     at ch.njol.skript.lang.SkriptParser.parse_i(SkriptParser.java:1474)
[18:09:25 ERROR]: #!#!     at ch.njol.skript.lang.SkriptParser.parse_i(SkriptParser.java:1513)
[18:09:25 ERROR]: #!#!     at ch.njol.skript.lang.SkriptParser.parse_i(SkriptParser.java:1474)
[18:09:25 ERROR]: #!#!     at ch.njol.skript.lang.SkriptParser.parse(SkriptParser.java:223)
[18:09:25 ERROR]: #!#!     at ch.njol.skript.lang.SkriptParser.parse(SkriptParser.java:178)
[18:09:25 ERROR]: #!#!     at ch.njol.skript.lang.Condition.parse(Condition.java:81)
[18:09:25 ERROR]: #!#!     at ch.njol.skript.ScriptLoader.loadItems(ScriptLoader.java:998)
[18:09:25 ERROR]: #!#!     at ch.njol.skript.lang.TriggerSection.<init>(TriggerSection.java:54)
[18:09:25 ERROR]: #!#!     at ch.njol.skript.lang.Conditional.<init>(Conditional.java:44)
[18:09:25 ERROR]: #!#!     at ch.njol.skript.ScriptLoader.loadItems(ScriptLoader.java:1005)
[18:09:25 ERROR]: #!#!     at ch.njol.skript.command.Commands.loadCommand(Commands.java:471)
[18:09:25 ERROR]: #!#!     at ch.njol.skript.ScriptLoader.loadScript(ScriptLoader.java:571)
[18:09:25 ERROR]: #!#!     at ch.njol.skript.ScriptLoader.lambda$1(ScriptLoader.java:347)
[18:09:25 ERROR]: #!#!     at ch.njol.skript.ScriptLoader.loadScripts(ScriptLoader.java:359)
[18:09:25 ERROR]: #!#!     at ch.njol.skript.ScriptLoader.lambda$0(ScriptLoader.java:289)
[18:09:25 ERROR]: #!#!     at ch.njol.skript.ScriptLoader.loadScripts(ScriptLoader.java:319)
[18:09:25 ERROR]: #!#!     at ch.njol.skript.Skript.reload(Skript.java:700)
[18:09:25 ERROR]: #!#!     at ch.njol.skript.SkriptCommand.onCommand(SkriptCommand.java:161)
[18:09:25 ERROR]: #!#!     at org.bukkit.command.PluginCommand.execute(PluginCommand.java:44)
[18:09:25 ERROR]: #!#!     at org.bukkit.command.SimpleCommandMap.dispatch(SimpleCommandMap.java:141)
[18:09:25 ERROR]: #!#!     at org.bukkit.craftbukkit.v1_12_R1.CraftServer.dispatchCommand(CraftServer.java:651)
[18:09:25 ERROR]: #!#!     at org.bukkit.craftbukkit.v1_12_R1.CraftServer.dispatchServerCommand(CraftServer.java:637)
[18:09:25 ERROR]: #!#!     at net.minecraft.server.v1_12_R1.DedicatedServer.aP(DedicatedServer.java:444)
[18:09:25 ERROR]: #!#!     at net.minecraft.server.v1_12_R1.DedicatedServer.D(DedicatedServer.java:407)
[18:09:25 ERROR]: #!#!     at net.minecraft.server.v1_12_R1.MinecraftServer.C(MinecraftServer.java:679)
[18:09:25 ERROR]: #!#!     at net.minecraft.server.v1_12_R1.MinecraftServer.run(MinecraftServer.java:577)
[18:09:25 ERROR]: #!#!     at java.base/java.lang.Thread.run(Unknown Source)
[18:09:25 ERROR]: #!#!
[18:09:25 ERROR]: #!#! Version Information:
[18:09:25 ERROR]: #!#!   Skript: 2.2-dev32b (latest)
[18:09:25 ERROR]: #!#!   Bukkit: 1.12.2-R0.1-SNAPSHOT
[18:09:25 ERROR]: #!#!   Minecraft: 1.12.2
[18:09:25 ERROR]: #!#!   Java: 9.0.1 (Java HotSpot(TM) 64-Bit Server VM 9.0.1+11)
[18:09:25 ERROR]: #!#!   OS: Windows 10 amd64 10.0
[18:09:25 ERROR]: #!#!
[18:09:25 ERROR]: #!#! Running CraftBukkit: false
[18:09:25 ERROR]: #!#! Running Spigot (or compatible): true
[18:09:25 ERROR]: #!#! Running Paper (or compatible): false
[18:09:25 ERROR]: #!#!
[18:09:25 ERROR]: #!#! Current node: arg 2 is not set: (clan.sk, line 5)
[18:09:25 ERROR]: #!#! Current item: null
[18:09:25 ERROR]: #!#! Thread: Server thread
[18:09:25 ERROR]: #!#!
[18:09:25 ERROR]: #!#! Language: english
[18:09:25 ERROR]: #!#! Link parse mode: DISABLED
[18:09:25 ERROR]: #!#!
[18:09:25 ERROR]: #!#! End of Error.
[18:09:25 ERROR]: #!#!
[18:09:25 ERROR]: #!#!
[18:09:25 ERROR]: #!#! [Skript] Severe Error:
[18:09:25 ERROR]: #!#! Could not load points.sk
[18:09:25 ERROR]: #!#!
[18:09:25 ERROR]: #!#! Something went horribly wrong with Skript.
[18:09:25 ERROR]: #!#! This issue is NOT your fault! You probably can't fix it yourself, either.
[18:09:25 ERROR]: #!#! It looks like you are using some plugin(s) that alter how Skript works (addons).
[18:09:25 ERROR]: #!#! Here is full list of them:
[18:09:25 ERROR]: #!#! skRayFall (http://sk.rayfall.net/) skUtilities (https://tim740.github.io/) SkQuery Skellett (https://forums.skunity.com/resources/skellett-the-addon-with-a-beast-name.24/) SharpSK ExertSK TuSKe (github.com/Tuke-Nuke/TuSKe)
[18:09:25 ERROR]: #!#! We could not identify which of those are specially related, so this might also be Skript issue.
[18:09:25 ERROR]: #!#! You should try disabling those plugins one by one, trying to find which one causes it.
[18:09:25 ERROR]: #!#! If the error doesn't disappear even after disabling all listed plugins, it is probably Skript issue.
[18:09:25 ERROR]: #!#! In that case, you will be given instruction on how should you report it.
[18:09:25 ERROR]: #!#! On the other hand, if the error disappears when disabling some plugin, report it to author of that plugin.
[18:09:25 ERROR]: #!#! Only if the author tells you to do so, report it as Skript issue (url below)
[18:09:25 ERROR]: #!#! Issue tracker: https://github.com/bensku/Skript/issues (only if you know what you're doing!)
[18:09:25 ERROR]: #!#!
[18:09:25 ERROR]: #!#! Stack trace:
[18:09:25 ERROR]: #!#! ch.njol.skript.SkriptAPIException: No class info found for lambda
[18:09:25 ERROR]: #!#!     at ch.njol.skript.registrations.Classes.getClassInfo(Classes.java:263)
[18:09:25 ERROR]: #!#!     at ch.njol.skript.lang.SkriptParser.createExprInfo(SkriptParser.java:1786)
[18:09:25 ERROR]: #!#!     at ch.njol.skript.lang.SkriptParser.getExprInfo(SkriptParser.java:1753)
[18:09:25 ERROR]: #!#!     at ch.njol.skript.lang.SkriptParser.parse_i(SkriptParser.java:1501)
[18:09:25 ERROR]: #!#!     at ch.njol.skript.lang.SkriptParser.parse(SkriptParser.java:223)
[18:09:25 ERROR]: #!#!     at ch.njol.skript.lang.SkriptParser.parse(SkriptParser.java:178)
[18:09:25 ERROR]: #!#!     at ch.njol.skript.lang.SkriptParser.parseSingleExpr(SkriptParser.java:491)
[18:09:25 ERROR]: #!#!     at ch.njol.skript.lang.SkriptParser.parseExpression(SkriptParser.java:808)
[18:09:25 ERROR]: #!#!     at ch.njol.skript.lang.SkriptParser.parse_i(SkriptParser.java:1517)
[18:09:25 ERROR]: #!#!     at ch.njol.skript.lang.SkriptParser.parse_i(SkriptParser.java:1474)
[18:09:25 ERROR]: #!#!     at ch.njol.skript.lang.SkriptParser.parse_i(SkriptParser.java:1513)
[18:09:25 ERROR]: #!#!     at ch.njol.skript.lang.SkriptParser.parse(SkriptParser.java:223)
[18:09:25 ERROR]: #!#!     at ch.njol.skript.lang.SkriptParser.parse(SkriptParser.java:178)
[18:09:25 ERROR]: #!#!     at ch.njol.skript.lang.Statement.parse(Statement.java:59)
[18:09:25 ERROR]: #!#!     at ch.njol.skript.ScriptLoader.loadItems(ScriptLoader.java:919)
[18:09:25 ERROR]: #!#!     at ch.njol.skript.command.Commands.loadCommand(Commands.java:471)
[18:09:25 ERROR]: #!#!     at ch.njol.skript.ScriptLoader.loadScript(ScriptLoader.java:571)
[18:09:25 ERROR]: #!#!     at ch.njol.skript.ScriptLoader.lambda$1(ScriptLoader.java:347)
[18:09:25 ERROR]: #!#!     at ch.njol.skript.ScriptLoader.loadScripts(ScriptLoader.java:359)
[18:09:25 ERROR]: #!#!     at ch.njol.skript.ScriptLoader.lambda$0(ScriptLoader.java:289)
[18:09:25 ERROR]: #!#!     at ch.njol.skript.ScriptLoader.loadScripts(ScriptLoader.java:319)
[18:09:25 ERROR]: #!#!     at ch.njol.skript.Skript.reload(Skript.java:700)
[18:09:25 ERROR]: #!#!     at ch.njol.skript.SkriptCommand.onCommand(SkriptCommand.java:161)
[18:09:25 ERROR]: #!#!     at org.bukkit.command.PluginCommand.execute(PluginCommand.java:44)
[18:09:25 ERROR]: #!#!     at org.bukkit.command.SimpleCommandMap.dispatch(SimpleCommandMap.java:141)
[18:09:25 ERROR]: #!#!     at org.bukkit.craftbukkit.v1_12_R1.CraftServer.dispatchCommand(CraftServer.java:651)
[18:09:25 ERROR]: #!#!     at org.bukkit.craftbukkit.v1_12_R1.CraftServer.dispatchServerCommand(CraftServer.java:637)
[18:09:25 ERROR]: #!#!     at net.minecraft.server.v1_12_R1.DedicatedServer.aP(DedicatedServer.java:444)
[18:09:25 ERROR]: #!#!     at net.minecraft.server.v1_12_R1.DedicatedServer.D(DedicatedServer.java:407)
[18:09:25 ERROR]: #!#!     at net.minecraft.server.v1_12_R1.MinecraftServer.C(MinecraftServer.java:679)
[18:09:25 ERROR]: #!#!     at net.minecraft.server.v1_12_R1.MinecraftServer.run(MinecraftServer.java:577)
[18:09:25 ERROR]: #!#!     at java.base/java.lang.Thread.run(Unknown Source)
[18:09:25 ERROR]: #!#!
[18:09:25 ERROR]: #!#! Version Information:
[18:09:25 ERROR]: #!#!   Skript: 2.2-dev32b (latest)
[18:09:25 ERROR]: #!#!   Bukkit: 1.12.2-R0.1-SNAPSHOT
[18:09:25 ERROR]: #!#!   Minecraft: 1.12.2
[18:09:25 ERROR]: #!#!   Java: 9.0.1 (Java HotSpot(TM) 64-Bit Server VM 9.0.1+11)
[18:09:25 ERROR]: #!#!   OS: Windows 10 amd64 10.0
[18:09:25 ERROR]: #!#!
[18:09:25 ERROR]: #!#! Running CraftBukkit: false
[18:09:25 ERROR]: #!#! Running Spigot (or compatible): true
[18:09:25 ERROR]: #!#! Running Paper (or compatible): false
[18:09:25 ERROR]: #!#!
[18:09:25 ERROR]: #!#! Current node: player has permission "skript.points.edit" (points.sk, line 4)
[18:09:25 ERROR]: #!#! Current item: null
[18:09:25 ERROR]: #!#! Thread: Server thread
[18:09:25 ERROR]: #!#!
[18:09:25 ERROR]: #!#! Language: english
[18:09:25 ERROR]: #!#! Link parse mode: DISABLED
[18:09:25 ERROR]: #!#!
[18:09:25 ERROR]: #!#! End of Error.
[18:09:25 ERROR]: #!#!

**Comment by bensku** [comment]
It does not seem that there is anything relevant concerning this issue anymore. It has been fixed for good.
